### PR TITLE
[TUNE] Tune Docs re-organization

### PR DESCRIPTION
**Pull request by bllchmbrs**
## Why are these changes needed?

The documentation for Tune is still a bit unwieldy - it's hard to know where files should go and what the "definitive" example for each thing should be.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by anabranch** [comment]
Screen Shot 2020-07-24 at 9 21 55 AM
Screen Shot 2020-07-24 at 9 21 47 AM
Screen Shot 2020-07-24 at 9 21 41 AM
Screen Shot 2020-07-24 at 9 21 35 AM
Screen Shot 2020-07-24 at 9 21 30 AM

File: doc/source/tune/_tutorials/overview.rst [hunk 1]
```diff
@@ -25,60 +21,36 @@ Take a look at any of the below tutorials to get started with Tune.
...
-   :description: :doc:`How to use Tune with PyTorch <tune-pytorch-cifar>`
+   :description: :doc:`How to use Tune with PyTorch <tutorials/tune-pytorch-cifar>`
 
 .. customgalleryitem::
    :tooltip: Tuning XGBoost parameters.
    :figure: /images/xgboost_logo.png
-   :description: :doc:`A guide to tuning XGBoost parameters with Tune <tune-xgboost>`
+   :description: :doc:`A guide to tuning XGBoost parameters with Tune <tutorials/tune-xgboost>`
+
+.. customgalleryitem::
```
Comment (amogkam) [inline doc/source/tune/_tutorials/overview.rst:45]: Would it be possible to add this PBT guide and the distributed experiments guide to the Tutorials dropdown on the left bar?

File: doc/source/tune/advanced.rst [hunk 2]
```diff
@@ -0,0 +1,467 @@
+Advanced Topics, Configurations, & FAQ
```
Comment (amogkam) [inline doc/source/tune/advanced.rst:1]: I'm not sure if Advanced Topics is the right name for this. Some of this stuff is pretty core and may not be solely for advanced users.

**Review by amogkam — COMMENTED** [review]
Thanks a lot for this! I also found some things difficult to navigate in the docs so this really helps. Just left a couple comments

File: doc/source/tune/_tutorials/overview.rst (hunk 1 above)
Comment (anabranch) [reply doc/source/tune/_tutorials/overview.rst:45]: Good idea.

File: doc/source/tune/advanced.rst (hunk 2 above)
Comment (anabranch) [reply doc/source/tune/advanced.rst:1]: What should I move out?  Do you have a recommendation?

Comment (richardliaw) [reply doc/source/tune/advanced.rst:1]: I would just rename this file, much of this isn't even advanced concepts - `User Guide and Configurations`?

Comment (amogkam) [reply doc/source/tune/advanced.rst:1]: +1 on just keeping it as `User Guide` or `Extended User Guide`

File: doc/source/tune/_tutorials/overview.rst [hunk 3]
```diff
@@ -88,13 +60,14 @@ These pages will demonstrate the various features and configurations of Tune.
...
-   tune-pytorch-lightning.rst
-   tune-xgboost.rst
+   tune-60-seconds.rst
+   tune-tutorial.rst
+   tutorials/tune-sklearn.rst
+   tutorials/tune-pytorch-cifar.rst
+   tutorials/tune-pytorch-lightning.rst
+   tutorials/tune-xgboost.rst
+   tutorials/tune-advanced-tutorial.rst
+   tutorials/tune-distributed.rst
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/overview.rst:70]: none of this is rendering properly?

**Comment by richardliaw** [comment]
The high level comments are:

1. I think we should split tutorials from examples
2. A lot of pages aren't rendering correctly (see comment)
3. Advanced Topics isn't an appropriate name for that page imo

Screen Shot 2020-07-27 at 11 56 28 AM

File: doc/source/tune/advanced.rst (hunk 2 above)
Comment (anabranch) [reply doc/source/tune/advanced.rst:1]: done

File: doc/source/tune/_tutorials/overview.rst (hunk 3 above)
Comment (anabranch) [reply doc/source/tune/_tutorials/overview.rst:70]: fixed I believe

**Review by richardliaw — APPROVED** [review]

**Pull request merged**